### PR TITLE
Add junit test result files to typescript functional and unit tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
#2830 

cc @natemcmaster

### CODE REVIEW

**Comment by anurse** [comment]
Urp, I accidentally pushed to your branch :). Just feel free to remove https://github.com/aspnet/SignalR/pull/2846/commits/251847861dba0afc563291a813ec22dfbb01cb72 entirely, it's not relevant.

**Comment by natemcmaster** [comment]
Re-triggering. This required https://github.com/aspnet/BuildTools/pull/743 which wasn't merged at the time this PR was opened.

**Comment by natemcmaster** [comment]
@mikaelm12 it looks like this change is only adding test results for typescript. But if there are Java test results too, it shouldn't be difficult to add those to the reporting.

**Comment by natemcmaster** [comment]
```
	Failed to read /Users/vsts/agent/2.136.1/work/1/s/artifacts/logs/signalr.junit.xml. Error : '', hexadecimal value 0x1E, is an invalid character. Line 421, position 63..
	Invalid results file. Make sure the result format of the file '/Users/vsts/agent/2.136.1/work/1/s/artifacts/logs/signalr.junit.xml' matches 'JUnit' test results format.
```

**Comment by mikaelm12** [comment]
I read JUnit and automatically thought that was related to Java... That explains my confusion. My bad 😄

**Comment by anurse** [comment]
Having said that @mikaelm12 , if you configure the Java Tests to emit a JUnit log file, with the extension `.junit.xml` to `artifacts/logs`, VSTS will pick them up now :)

File: clients/ts/FunctionalTests/package.json [hunk 1]
```diff
@@ -11,19 +11,20 @@
...
+    "@types/node": "^9.6.28",
     "debug": "^3.1.0",
-    "es6-promise": "^4.2.2",
-    "jasmine": "^3.1.0",
-    "jasmine-core": "^3.1.0",
-    "karma": "^2.0.4",
+    "es6-promise": "^4.2.4",
+    "jasmine": "^3.2.0",
+    "jasmine-core": "^3.2.1",
+    "karma": "^3.0.0",
```
Comment (anurse) [inline clients/ts/FunctionalTests/package.json]: @Eilon Note: Updated dependency. Still used at test time only. I'll capture this in my work on automating registrations.

**Comment by BrennanConroy** [comment]
@anurse Take a quick look at this. Now with escaping

**Comment by natemcmaster** [comment]
LGTM. :shipit:

image

File: clients/ts/package.json [hunk 2]
```diff
@@ -19,6 +19,10 @@
         "enableTsDiagnostics": true
       }
     },
+    "reporters": [
+      "default",
+      [ "./common/node_modules/jest-junit/index.js", { "output": "../../artifacts/logs/signalr.junit.xml" } ]
```
Comment (anurse) [inline clients/ts/package.json]: Use a `.js` config file (https://jestjs.io/docs/en/configuration.html) and insert the OS into the file name (`../../artifacts/logs/signalr.unit-tests.${process.platform}.junit.xml`)

Ditto for karma, and maybe see what the XML looks like so we can try to get all the browsers to show up properly?

**Pull request merged**